src/components/flyout/flyout.styles.ts

```diff
@@ -0,0 +1,282 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { css, keyframes } from '@emotion/react';
+import { EuiFlyoutPaddingSize } from './flyout';
+import { euiCanAnimate, euiBreakpoint, logicalCSS } from '../../global_styling';
+import { UseEuiTheme } from '../../services';
+import { euiShadowXLarge } from '../../themes/amsterdam/global_styling/mixins';
+import { transparentize } from '../../services/color';
+import { euiFormMaxWidth } from '../form/form.styles';
+
+const euiFlyout = keyframes`
+  0% {
+    opacity: 0;
+    transform: translateX(100%);
+  }
+  75% {
+    opacity: 1;
+    transform: translateX(0%);
+  }
+`;
+
+const euiFlyoutLeft = keyframes`
+  0% {
+    opacity: 0;
+    transform: translateX(-100%);
+  }
+  75% {
+    opacity: 1;
+    transform: translateX(0%);
+}
+`;
+
+export const euiFlyoutCloseButtonStyles = (euiThemeContext: UseEuiTheme) => {
+  const euiTheme = euiThemeContext.euiTheme;
+
+  return {
+    euiFlyout__closeButton: css`
+      background-color: ${transparentize(euiTheme.colors.emptyShade, 0.1)};
+      position: absolute;
+      ${logicalCSS('right', euiTheme.size.s)}
+      ${logicalCSS('top', euiTheme.size.s)}
+      z-index: 3;
+    `,
+    'closeButton--outside': css`
+      // match dropshadow
+      ${euiShadowXLarge(euiThemeContext)};
+      ${logicalCSS('right', 'auto')}
+      ${logicalCSS('left', 0)}
+      // Override the hover and focus transitions of buttons
+      animation: none !important;
+      ${euiBreakpoint(euiThemeContext, ['m', 'xl'])} {
+        transform: translateX(calc(-100% - ${euiTheme.size.l}));
+      }
+      ${euiBreakpoint(euiThemeContext, ['xs', 's'])} {
+        transform: translateX(calc(-100% - ${euiTheme.size.xs}));
+      }
+    `,
+    'closeButton--inside': css``,
+    'closeButton--outside-left': css`
+      ${logicalCSS('right', 0)}
+      ${logicalCSS('left', 'auto')}
+      
+      ${euiBreakpoint(euiThemeContext, ['m', 'xl'])} {
+        transform: translateX(calc(100% + ${euiTheme.size.l}));
+      }
+      ${euiBreakpoint(euiThemeContext, ['xs', 's'])} {
+        transform: translateX(calc(100% + ${euiTheme.size.xs}));
+      }
+    `,
+  };
+};
+
+export const euiFlyoutStyles = (
+  euiThemeContext: UseEuiTheme,
+  paddingSize: EuiFlyoutPaddingSize
+) => {
+  const euiTheme = euiThemeContext.euiTheme;
+
+  const euiFormMaxWidthNumber: number = parseInt(
+    euiFormMaxWidth(euiThemeContext).replace('px', ''),
+    10
+  );
```

Comment:
see above comment re parseInt/replace
```suggestion
  const euiFormMaxWidthNumber = parseInt(euiFormMaxWidth(euiThemeContext), 10);
```